torch/csrc/jit/script/schema_matching.cpp

```diff
@@ -84,13 +83,10 @@ Value* tryConvertToType(
 
   if (value->type()->isSubtypeOf(NoneType::get()) &&
       !concrete_type->isSubtypeOf(NoneType::get())) {
-    if (concrete_type->isSubtypeOf(OptionalType::ofTensor())) {
-      // create undefined tensor when None pass to a optional[tensor] formal arg
-      value = graph.insertNode(graph.createUndefined())->output();
-    } else if (auto optional_type = concrete_type->cast<OptionalType>()) {
-      value =
-          graph.insertNode(graph.createNone(optional_type->getElementType()))
-              ->output();
+    if (concrete_type->cast<OptionalType>()) {
+      value->setType(concrete_type);
+    } else {
+      value->setType(OptionalType::create(concrete_type));
```

Comment:
This change is wrong. Again, you can't change the type of a Value. Values can be used in more than one place. The right code is what was there before.